mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,868 @@
+        """
+        dim = feat.size(2)
+        ind = ind.unsqueeze(2).expand(ind.size(0), ind.size(1), dim)
+        feat = feat.gather(1, ind)
+        if mask is not None:
+            mask = mask.unsqueeze(2).expand_as(feat)
+            feat = feat[mask]
+            feat = feat.view(-1, dim)
+        return feat
+
+    def get_targets(self, gt_bboxes_3d, gt_labels_3d):
+        """Generate targets.
+
+        Args:
+            gt_bboxes_3d (list[:obj:`LiDARInstance3DBoxes`]): Ground
+                truth gt boxes.
+            gt_labels_3d (list[torch.Tensor]): Labels of boxes.
+
+        Returns:
+            list[torch.Tensor]: Heatmap scores.
+            list[torch.Tensor]: Ground truth boxes.
+            list[torch.Tensor]: Indexes indicating the
+                position of the valid boxes.
+            list[torch.Tensor]: Masks indicating which boxes are valid.
+        """
+        heatmaps, anno_boxes, inds, masks = multi_apply(
+            self.get_targets_single, gt_bboxes_3d, gt_labels_3d)
+
+        # transpose heatmaps
+        heatmaps_transed = [[] for _ in range(6)]
+        for i in range(len(heatmaps)):
+            for j in range(len(heatmaps[i])):
+                heatmaps_transed[j].append(heatmaps[i][j])
+        heatmaps = [torch.stack(heatmaps_) for heatmaps_ in heatmaps_transed]
+
+        # transpose anno_boxes
+        anno_boxes_transed = [[] for _ in range(6)]
+        for i in range(len(anno_boxes)):
+            for j in range(len(anno_boxes[i])):
+                anno_boxes_transed[j].append(anno_boxes[i][j])
+        anno_boxes = [
+            torch.stack(anno_boxes_) for anno_boxes_ in anno_boxes_transed
+        ]
+
+        # transpose inds
+        inds_transed = [[] for _ in range(6)]
+        for i in range(len(inds)):
+            for j in range(len(inds[i])):
+                inds_transed[j].append(inds[i][j])
+        inds = [torch.stack(inds_) for inds_ in inds_transed]
+        masks = np.array(masks).transpose(1, 0).tolist()
+        masks = [torch.stack(masks_) for masks_ in masks]
+        return heatmaps, anno_boxes, inds, masks
+
+    def get_targets_single(self, gt_bboxes_3d, gt_labels_3d):
+        """Generate training targets for a single sample.
+
+        Args:
+            gt_bboxes_3d (:obj:`LiDARInstance3DBoxes`): Ground truth gt boxes.
+            gt_labels_3d (torch.Tensor): Labels of boxes.
+
+        Returns:
+            list[torch.Tensor]: Heatmap scores.
+            list[torch.Tensor]: Ground truth boxes.
+            list[torch.Tensor]: Indexes indicating the position
+                of the valid boxes.
+            list[torch.Tensor]: Masks indicating which boxes
+                are valid.
+        """
+        device = gt_labels_3d.device
+        gt_bboxes_3d = torch.cat((gt_bboxes_3d.gravity_center,
+                                  gt_bboxes_3d.tensor[:, [3, 4, 5, 7, 8, 6]]),
+                                 dim=1).to(device)
+        gt_labels_3d += 1
```

Comment:
Why do we need to add 1 here?

Reply:
Our label is 1 less than the source code of CenterPoint.

Reply:
This is because our in MMDet3D and MMDet, 0-N-1 is foreground labels for N classes and N is for background class. Please make them consistent in the system.